frontend/src/lib/k8s/cluster.ts

```diff
@@ -363,35 +363,52 @@ type JsonPath<T> = T extends object
  * @param objectName The name of the object to create a KubeObject implementation for.
  */
 export function makeKubeObject<T extends KubeObjectInterface | KubeEvent>(
+  // eslint-disable-next-line no-unused-vars -- not used for now, to be removed
   objectName: string
 ): KubeObjectIface<T> {
   class KubeObject {
     static apiEndpoint: ReturnType<typeof apiFactoryWithNamespace | typeof apiFactory>;
     jsonData: T | null = null;
-    public static readOnlyFields: JsonPath<T>[];
+    /** Readonly field defined as JSONPath paths */
+    static readOnlyFields: JsonPath<T>[] = [];
     private readonly _clusterName: string;
 
+    /** The kind of the object. Corresponding to the resource kind in Kubernetes. */
+    static readonly kind: string;
+
+    /** Name of the resource, plural, used in API */
+    static readonly apiName: string;
+
+    /** Group and version of the resource formatted as "GROUP/VERSION", e.g. "policy.k8s.io/v1". */
+    static readonly apiVersion: string | string[];
+
+    /** Whether the object is namespaced. */
+    static readonly isNamespaced: boolean;
+
+    /** The former name of this class in Headlamp. Soon to be deprecated, use kind instead. */
+    static readonly legacyObjectName?: string;
```

Comment:
I didn't see this being used for anything, but I think we need to use it when adding the classes to the ResourceClasses.
And if possible, the ResourceClasses should override its accessor so when users access ResourceClasses['ingress'], it prints a deprecation statement that it's a "legacy name for the class, please use Ingress".

Reply:
ResourceClasses key is the same as `kind` (except for Endpoint), it's capitalized manually when constructing the ResourceClasses object. We don't really need legacyObjectName